onnxmltools/convert/common/_topology.py

```diff
@@ -593,18 +594,26 @@ def compile(self):
         self._check_structure()
 
 
-def convert_topology(topology, model_name, doc_string):
+def convert_topology(topology, model_name, doc_string, targeted_onnx):
     '''
     This function is used to convert our Topology object defined in _parser.py into a ONNX model (type: ModelProto).
     :param topology: The Topology object we are going to convert
-    :param model_name: GraphProto's name. Let "model" denote the output model. The string "model_name" would be assigned
-    to "model.graph.name."
+    :param model_name: GraphProto's name. Let "model" denote the returned model. The string "model_name" would be
+    assigned to "model.graph.name."
     :param doc_string: A string attached to the produced model
+    :param targeted_onnx: A string, which specifies the targeted ONNX version of the produced model. Possible values
+    include '1.1.2', '1.2', and so on.
     :return: a ONNX ModelProto
     '''
+    if targeted_onnx != onnx.__version__:
+        raise RuntimeError(
+            'ONNX version conflict found. The installed version is %s while the targeted version is %s' % (
+                targeted_onnx, onnx.__version__))
```

Comment:
This is much better, but we may be able to do this earlier. Good for this PR though.

Reply:
Alternatively, we can this this in convert(…), the function user calls to convert a model. It means that there will be three copies of the same check for Core ML, scikit-learn, and Keras. Although checking ONNX version here is a bit late in a conversion pipeline, it is more manageable to me.